### PR TITLE
Chore/tt2 412 subcribe logging bucket to csls

### PR DESCRIPTION
**Pull request by gds-topickering**

### CODE REVIEW

File: core/template.yaml [hunk 1]
```diff
@@ -301,6 +301,43 @@ Resources:
       AliasName: !Sub alias/${AWS::StackName}/${Environment}/queue-kms-key
       TargetKeyId: !Ref SqsKmsKey
 
+  CslsKmsKey:
```
Comment (andylangridge) [inline core/template.yaml:304]: is this just a generic key to encrypt s3 buckets? If so, then we could we make this a generic key? ie. call it `S3KmsKey`?

File: core/template.yaml [hunk 2]
```diff
@@ -367,11 +411,58 @@ Resources:
...
+              - 'sqs:ChangeMessageVisibility'
+              - 'sqs:DeleteMessage'
+              - 'sqs:GetQueueAttributes'
+              - 'sqs:ReceiveMessage'
+            Effect: 'Allow'
+            Resource: !GetAtt S3CslsQueue.Arn
+            Principal:
+              AWS: '{{resolve:ssm:CSLSS3LambdaArn}}'
+
+  S3CslsDeadLetterQueue:
```
Comment (andylangridge) [inline core/template.yaml:460]: Should we also have an alert when something lands in this queue?

File: core/template.yaml [hunk 3]
```diff
@@ -367,11 +411,58 @@ Resources:
...
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${AWS::StackName}-${Environment}-s3-csls-dead-letter-queue
+      KmsMasterKeyId: !GetAtt CslsKmsKey.Arn
```
Comment (andylangridge) [inline core/template.yaml:464]: Can we make the message retention period 14 days as well? The default is only 4.
```suggestion
      KmsMasterKeyId: !GetAtt CslsKmsKey.Arn
      MessageRetentionPeriod: 1209600
```

Comment (gds-topickering) [reply core/template.yaml:464]: Done

File: core/template.yaml (hunk 2 above)
Comment (gds-topickering) [reply core/template.yaml:460]: I've added an alarm

**Review by andylangridge — APPROVED** [review]

**Pull request merged**